### PR TITLE
Add --args to MooseDocs build command for overriding config

### LINKED ISSUE #16807: Allow MooseDocs config edits from command-line
## Reason

When deploying documentation certain features may need to be disabled (e.g., complete source code links when using listing command). 

## Design

Add command-line argument that overrides supplied config items. The argument should be a string to be parsed by the YAML loader.

```
./moosedocs.py --args "{Extensions: {MooseDocs.extensions.listing" {modal-link: 0}}}"
```

## Impact

Improved control of documentation for deployment.

### PR DESCRIPTION
**Pull request by aeslaughter**
This is for NCRC deployment work.
(closes #16807)

### CODE REVIEW

File: python/MooseDocs/test/commands/tests [hunk 1]
```diff
@@ -22,4 +22,11 @@
 
     requirement = "The system shall include a utility for displaying the application syntax tree."
   []
+  [build]
+    type = PythonUnitTest
+    input = test_build.py
+    issues = '#16807'
+
+    requirement = "The system shall include a utility for converting markdown documentation to other formats."
```
Comment (loganharbour) [inline python/MooseDocs/test/commands/tests:30]: I don't believe this is right...

**Review by loganharbour — DISMISSED** [review]

**Comment by aeslaughter** [comment]
There is another option that I need to make sure I can turn off, 'SourceLink' tokens. Then document the ability to turn things off.

File: python/MooseDocs/test/commands/tests (hunk 1 above)
Comment (aeslaughter) [reply python/MooseDocs/test/commands/tests:30]: It is, this is testing the "build" command. Which does the converting.

**Review by rwcarlsen — APPROVED** [review]
I confirmed that there were tests.  I didn't see anything stand out as wrong.  Although I'm really not familiar with moose docs code at all.

**Pull request merged**